matchzoo/engine/base_model.py

```diff
@@ -186,8 +189,31 @@ def evaluate(
             give you the display labels for the scalar outputs.
 
         """
-        return self._backend.evaluate(x=x, y=y,
-                                      batch_size=batch_size, verbose=verbose)
+        backend_evals = self._backend.evaluate(x=x, y=y,
+                                               batch_size=batch_size,
+                                               verbose=verbose)
+        if not isinstance(backend_evals, list):
+            backend_evals = [backend_evals]
+        metrics_lookup = {name: val for name, val in
+                          zip(self._backend.metrics_names, backend_evals)}
+        list_wise_df = None
+        for metric in self._params['task'].metrics:
+            if isinstance(metric, engine.BaseMetric):
+                if list_wise_df is None:
+                    y_pred = self.predict(x, batch_size)
```

Comment:
Here, we can adjust the code to improve efficiency. The line `y_pred = self.predict(x, batch_size)` can be moved out of the loop `for'.

Reply:
It only happens once already.